Floofbot/Modules/Administration.cs

```diff
@@ -506,6 +510,280 @@ public async Task warnlog([Summary("user")] string user)
+                return;
+            }
+
+            if (type == "warning") // forgive warning
+            {
+                if (!_floofDB.Warnings.AsQueryable().Where(w => w.UserId == uID && w.GuildId == Context.Guild.Id && w.Forgiven == false).Any()) // there are no warnings for this user in this guild
+                {
+                    await Context.Channel.SendMessageAsync("User has no warnings to forgive!");
+                    return;
+                }
+                warnings = _floofDB.Warnings.AsQueryable()
+                    .Where(u => u.UserId == uID && u.GuildId == Context.Guild.Id && u.Forgiven == false)
+                    .OrderByDescending(x => x.DateAdded).Take(10);
+            }
+            else if (type == "usernote") // forgive usernote
+            {
+                if (!_floofDB.UserNotes.AsQueryable().Where(w => w.UserId == uID && w.GuildId == Context.Guild.Id && w.Forgiven == false).Any()) // there are no user notes for this user in this guild
+                {
+                    await Context.Channel.SendMessageAsync("User has no user notes to forgive!");
+                    return;
+                }
+                warnings = _floofDB.UserNotes.AsQueryable()
+                    .Where(u => u.UserId == uID && u.GuildId == Context.Guild.Id && u.Forgiven == false)
+                    .OrderByDescending(x => x.DateAdded).Take(10);
+            }
+
+            EmbedBuilder builder = new EmbedBuilder();
+            builder.Color = ADMIN_COLOR;
+            if (user == null) // no user, just id in database
+                builder.WithTitle($"Warnings for {badUser}");
+            else
+                builder.WithTitle($"Warnings for {user.Username}#{user.Discriminator}");
+            if (warnings == null) // for some reason didnt recieve data from database
+            {
+                Log.Error("Fatal error when trying to access warnings for the forgive user command!");
+                return;
+            }
+            if (type == "warning")
+            {
+                foreach (Warning w in warnings)
+                {
+                    builder.AddField($"**ID: {w.Id}** - {w.DateAdded.ToString("yyyy MMMM dd")} - {w.Moderator}", $"```{w.Reason}```");
+                }
+            }
+            else if (type == "usernote")
+            {
+                foreach (UserNote w in warnings)
+                {
+                    builder.AddField($"**ID: {w.Id}** - {w.DateAdded.ToString("yyyy MMMM dd")} - {w.Moderator}", $"```{w.Reason}```");
+                }
+            }
+            await SendEmbed(builder.Build());
+            try
+            {
+                await ReplyAsync("Which would you like to forgive? Please specify the ID.");
+
+                SocketMessage response = await NextMessageAsync(true, true, TimeSpan.FromSeconds(10)); // wait for reply from source user in source channel for 2 seconds
```

Comment:
The earlier comment was probably lost, but here the code says 10 seconds, but your comment says 2 seconds. Which one did you want?

Reply:
Same thing is in the unforgive function.

Reply:
Fixed